Kudu.Services/ServiceHookHandlers/GithubHandler.cs

```diff
@@ -17,6 +36,25 @@ public override DeployAction TryParseDeploymentInfo(HttpRequestBase request, JOb
             return DeployAction.UnknownPayload;
         }
 
+        protected override string determineSecurityProtocol(JObject repository)
+        {
+            string repositoryUrl = repository.Value<string>("url");
+            bool isPrivate = repository.Value<bool>("private");
+            if (isPrivate)
+            {
+                repositoryUrl = repository.Value<string>("ssh_url");
```

Comment:
Why use different logic in base and derived to get the ssh url? I'd suggest adding a `GetSSHUrl` helper in the base and use it from all 3 places.